src/ShallowWrapper.js

```diff
@@ -346,16 +347,22 @@ export default class ShallowWrapper {
    * @param {Array} args
    * @returns {ShallowWrapper}
    */
-  simulate(event, ...args) {
-    const handler = this.prop(propFromEvent(event));
-    if (handler) {
-      withSetStateAllowed(() => {
-        // TODO(lmr): create/use synthetic events
-        // TODO(lmr): emulate React's event propagation
-        handler(...args);
-        this.root.update();
-      });
-    }
+  simulate(event, mock, ...args) {
+    const eventProp = propFromEvent(event);
+    const e = Object.assign(new SyntheticEvent(undefined, undefined, { type: event }), mock);
```

Comment:
if the mock replaces the `stopPropagation` method event propagation stops working as intended so this `Object.assign` is probably not a great idea